Frontend/src/components/Register.js

```diff
@@ -0,0 +1,245 @@
+import React, { useState } from "react";
+import { useForm } from 'react-hook-form';
+import {useHistory} from "react-router-dom";
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+
+import AuthService from "../services/auth.service";
+
+const Register = () => {
+  const history = useHistory();
+  const [successful, setSuccessful] = useState(false);
+  const [message, setMessage] = useState("");
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string()
+        .required('Name is required')
+        .min(10, 'Name must be at least 10 characters')
+        .max(30, 'Name must not exceed 30 characters'),
+    surname: Yup.string()
+        .required('Surname is required')
+        .min(10, 'Surname must be at least 10 characters')
+        .max(30, 'Surname must not exceed 30 characters'),
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/JeremiasFuentes/unq-desapp-fuentes-diaz/pullRequest?prid=8198099)
```suggestion
        .max(30, "Surname must not exceed 30 characters"),
```